drivers/serial/uart_sam0.c

```diff
@@ -778,12 +786,79 @@ static int uart_sam0_fifo_read(const struct device *dev, uint8_t *rx_data,
 
 static int uart_sam0_irq_is_pending(const struct device *dev)
 {
-	SercomUsart *const regs = DEV_CFG(dev)->regs;
+	SercomUsart * const regs = DEV_CFG(dev)->regs;
 
 	return (regs->INTENSET.reg & regs->INTFLAG.reg) != 0;
 }
 
-static int uart_sam0_irq_update(const struct device *dev) { return 1; }
+#if defined(SERCOM_REV500)
+static void uart_sam0_irq_err_enable(const struct device *dev)
+{
+	SercomUsart * const regs = DEV_CFG(dev)->regs;
+
+	regs->INTENSET.reg |= SERCOM_USART_INTENCLR_ERROR;
+	wait_synchronization(regs);
+}
+
+static void uart_sam0_irq_err_disable(const struct device *dev)
+{
+	SercomUsart * const regs = DEV_CFG(dev)->regs;
+
+	regs->INTENCLR.reg |= SERCOM_USART_INTENSET_ERROR;
+	wait_synchronization(regs);
+}
+#endif
```

Comment:
This block should be after `uart_sam0_err_check`

Reply:
Why? It relates to irq and it is right after `uart_sam0_irq_is_pending` and other irq functions.

Reply:
In general, not a strong rule, blocks goes after common code. I looked at the driver and saw many blocks and they are spread, so, let's keep as you defined.

Reply:
Thank you. I'll keep it in mind for future PRs.